src/Migration/Exception.php

```diff
@@ -4,18 +4,26 @@
 
 class Exception extends \Exception
 {
+    public string $resourceName;
+
     public string $resourceType;
 
     public string $resourceId;
 
-    public function __construct(string $resourceType, string $message, int $code = 0, ?\Throwable $previous = null, string $resourceId = '')
+    public function __construct(string $resourceName, string $resourceType, string $message, int $code = 0, ?\Throwable $previous = null, string $resourceId = '')
```

Comment:
Not sure about name of variables. If one is Storage and one is Bucket, I think we should do $service and $resource

Reply:
Migration lib is supposed to be agnostic so it's all called groups throughout the library, changing it here would mean we would have to change it everywhere. We could probably change the terminology for G2 but I think we should keep it the same for G1